td-uefi-pi/src/hob.rs

```diff
@@ -49,6 +49,7 @@ pub fn check_hob_length(hob: &[u8], hob_length: usize) -> Option<&[u8]> {
     if phit.header.r#type == HOB_TYPE_HANDOFF
         && phit.header.length as usize >= size_of::<HandoffInfoTable>()
     {
+        phit.efi_end_of_hob_list.checked_sub(hob.as_ptr() as u64)?;
```

Comment:
I think we just need sub once. No need to sub twice.

Reply:
Fixed